.github/workflows/deployment.yaml

```diff
@@ -0,0 +1,121 @@
+name: "Deployment"
+
+on:
+  workflow_call:
+    inputs:
+      deployment_directory:
+        required: true
+        type: string
+
+      deployment_file:
+        required: false
+        type: string
+        default: "values.yaml"
+
+      service_name:
+        required: false
+        type: string
+        default: "ltms-demo"
+
+      environment:
+        required: true
+        type: string
+
+      repository:
+        required: false
+        type: string
+        default: kargotech/rhodes
+
+      gcp_project_id:
+        required: false
+        type: string
+        default: kargotech-prod
+
+      gke_cluster:
+        required: false
+        type: string
+        default: helios
+
+      gke_zone:
+        required: false
+        type: string
+        default: asia-southeast2-a
+
+    secrets:
+      deployer_app_id:
+        required: true
+      deployer_private_key:
+        required: true
+      deployer_installation_id:
+        required: true
+
+      gke_sa_key:
+        required: true
+
+jobs:
+  deployment:
+    name: Deploy
+    runs-on: ubuntu-20.04
+    environment: ${{ inputs.environment }}
+
+    steps:
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v1
+        with:
+          app_id: ${{ secrets.deployer_app_id }}
+          private_key: ${{ secrets.deployer_private_key }}
+          installation_id: ${{ secrets.deployer_installation_id }}
+          repository: ${{ inputs.repository }}
+
+      - name: Checkout code
```

Comment:
Why we need to checkout repo code?

Reply:
to get Author information. I try to make commit in target repo similar to source commit